### DISCUSSION TITLE
Pending pengiriman terasa lebih lama dan respon webhook sent_message_response berbeda

### DISCUSSION THREAD

**Discussion by masbagusstwn** [kirimwa-id/forum#154] [Tanya Jawab]
Siang mas rio,

Saya sangat mengapresiasi adanya api kirimWA ini sangat-sangat membantu dan memudahkan saya dalam mengelola pesan masuk dari whatsapp.

Saya ingin tanyakan untuk proses pengiriman pesan sekarang terasa sedikit lebih lama dari sebelum-sebelumnya ya mas rio? apakah memang karena pengguna api kirimWA yang sudah cukup banyak ditambah trafic kirim pesan yang makin meningkat sehingga menyebabkan proses sedikit lebih lama dari sebelum2nya ??

Ditambah lagi ketika proses berhasil, namun response dari sent_message_response berbeda dari apa yg telah saya kirim. [terlampir dengan message_idnya).

Sudah lebih dari 1 kali saya mengalami hal demikian. Untuk pesan yg terkirim sudah sesuai dan terkirim dengan benar tetapi hasil dari sent_message_responsenya yg berbeda seperti gambar yang saya lampirkan.

Sedikit saran, mungkin bisa di batasin saja mas kuota pengiriman pesannya per token_id atau dibatasi untuk requestnya tidak bisa berapa kali request dalam satu menit (seperti halnya api lain yang memiliki flooding). Karena jika saya baca dari respon salah yg saya terima ini mirip-mirip dijadikan notifikasi bukan digunakan untuk keperluan yang memang untuk mengirim pesan / membalas pesan seperti pada umumnya.

Catatan. : Lebih baik yg memang digunakan untuk keperluan notifikasi atau feedback respon diarahkan menggunakan yang berbisnis / berbayar saja. Jangan yang gratisan. Kasihan developer lain yang memang menggunakan API free ini untuk kepentingan development test

Terimakasih,

![image](https://user-images.githubusercontent.com/96308070/203712658-492b1aab-2abc-4e81-abb5-69a6d7932a1f.png)

**Comment by masbagusstwn** [comment]
terjadi lagi, dari notifikasi yg sama lagi pada pagi ini.

**Reply by rioastamal** [reply]
Halo mas @masbagusstwn, mohon maaf atas atas ketidaknyamanannya. Betul jumlah user sudah semakin banyak 1500+. Ada beberapa variabel pemicu dan banyaknya user adalah salah satunya disamping keperluan untuk optimasi lebih lanjut.

API KirimWA.id rencana akan resmi mengeluarkan paket berbayar per bulan Maret 2023. Hal ini dilakukan untuk mengakomodir user yang memiliki keperluan bisnis.

Sementara itu, optimasi akan terus dilakukan. Kurang lebih seminggu kedepan saya masih berada di Las Vegas, jadi mungkin akan slow response.

Silahkan email mesage_id yang nyasar tersebut dan terindikasi spam. Thanks.

**Comment by masbagusstwn** [comment]
Baik mas secara teknisnya saya sedikit mengerti jika demikian.

Terkait API kirimWA.id yang berbisnis sebetulnya lebih ke saran
pribadi saya untuk pembaca thread ini jika memang memerlukan untuk
kepentingan broadcasting / notifikasi alangkah baiknya nantinya supaya
menggunakan versi bussiness saja sehingga tidak membebani versi free
nya sekarang ini.

Karena memang baru kali ini dan mungkin baru ini saya menemukan
penggunaan pengolah whatsapp berbasis API  yang dapat di gunakan
dengan menggunakan fetch URL (karena saya intens menggunakan Google
Apps Script)

Mohon maaf sebelumnya ya mas Rio jika terdapat salah kata.

Nanti akan sy coba kelompokkan kembali jika terdapat kesalahan respon
ketika mengirim menggunakan api kirimWA.

Sukses terus mas Rio dengan API kirimWA.id nya ������
